src/main/java/de/thomaskrille/dropwizard_template_config/TemplateConfigBundleConfiguration.java

```diff
@@ -125,4 +133,12 @@ public TemplateConfigBundleConfiguration outputPath(String outputPath) {
         this.outputPath = Optional.of(outputPath);
         return this;
     }
+
+    /**
+     * Add a custom provider used to add your own variables to the configuration template.
+     */
+    public TemplateConfigBundleConfiguration withCustomProvider(TemplateConfigProvider customProvider) {
```

Comment:
I think it's better to call this method `addCustomProvider`. This way, you instantly see that the method adds a new provider. The prefix `with` does not carry this information.